src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmParticipantSpec.scala

```diff
@@ -0,0 +1,399 @@
+  private val superiorGridAgent = SuperiorGA(
+    TestProbe("superiorGridAgent_1000"),
+    Seq(supNodeA.getUuid)
+  )
+
+  "Test participant" should {
+
+    val gridAgentWithParticipants = system.actorOf(
+      GridAgent.props(
+        environmentRefs,
+        simonaConfig,
+        Iterable(resultListener.ref)
+      )
+    )
+
+    s"initialize itself when it receives a $InitializeGridAgentTrigger with corresponding data" in {
+      val triggerId = 0
+      val loadAgentTriggerId = 1
+
+      // this subnet has 1 superior grid (HöS) and 3 inferior grids (MS). Map the gates to test probes accordingly
+      val subGridGateToActorRef: Map[SubGridGate, ActorRef] =
+        hvSubGridGates.map {
+          case gate if gate.getInferiorSubGrid == hvGridContainer.getSubnet =>
+            gate -> superiorGridAgent.ref
+        }.toMap
+
+      val gridAgentInitData =
+        GridAgentInitData(
+          hvGridContainer,
+          subGridGateToActorRef,
+          RefSystem("2000 MVA", "110 kV")
+        )
+
+      // send init data to agent and expect a CompletionMessage
+      scheduler.send(
+        gridAgentWithParticipants,
+        TriggerWithIdMessage(
+          InitializeGridAgentTrigger(gridAgentInitData),
+          triggerId,
+          gridAgentWithParticipants
+        )
+      )
+
+      val (loadAgent, initializeTrigger): (
+          ActorRef,
+          InitializeParticipantAgentTrigger[PrimaryData, InitializeStateData[
+            PrimaryData
+          ]]
+      ) =
+        scheduler.expectMsgPF() {
+          case ScheduleTriggerMessage(
+                initializeTrigger: InitializeParticipantAgentTrigger[
+                  PrimaryData,
+                  InitializeStateData[PrimaryData]
+                ],
+                loadAgent
+              ) =>
+            (loadAgent, initializeTrigger)
+        }
+
+      scheduler.expectMsgPF() {
+        case CompletionMessage(
+              triggerId,
+              Some(
+                Vector(
+                  ScheduleTriggerMessage(triggerToBeScheduled, gridAgentActor)
+                )
+              )
+            ) =>
+          triggerId shouldBe 0
+          triggerToBeScheduled shouldBe ActivityStartTrigger(3600L)
+          gridAgentActor shouldBe gridAgentWithParticipants
+        case x =>
+          fail(
+            s"Invalid message received when expecting a completion message after an init trigger. Message was $x"
+          )
+      }
+
+      scheduler.send(
+        loadAgent,
+        TriggerWithIdMessage(
+          initializeTrigger,
+          loadAgentTriggerId,
+          loadAgent
+        )
+      )
+
+      primaryService.expectMsg(
+        PrimaryServiceRegistrationMessage(load1.getUuid)
+      )
+
+      primaryService.send(loadAgent, RegistrationFailedMessage)
+
+      scheduler.expectMsgPF(10.seconds) {
+        case CompletionMessage(
+              loadAgentTriggerId,
+              Some(triggerMessage :: Nil)
+            ) =>
```

Comment:
I know the idea of constructing the list with `Nil` came from me, but I'd rather have the test of the received message via a `expectMsg`, where we test for `CompletionMessage(loadAgentTriggerId, Some(Seq(ActivityStartTrigger(0L))))` directly.